### DISCUSSION TITLE
Trying to change colors of image with different antialiased colors

### DISCUSSION THREAD

**Discussion by oakif** [ImageMagick/ImageMagick#2876] [Help] [answered]
I'm trying to replace the colors of an image. Essentially I'm trying to make the background (`#FFFFFF`) into a gray color (approximately `#222222`). However, on this image there is both black text and text of other colors (blue, yellow, orange, etc). A sample image is shown below: 

![image](https://user-images.githubusercontent.com/48932412/100493168-72c2d980-3102-11eb-9fad-3740cd5d0a65.png)

It's probably easy to see why this such a difficult task. I tried replacing the black first with the `fuzz` option. However, from my understanding this doesn't _retain_ fuzziness, causing harsh edges that should be antialiased:

Command:
`convert in.png -fuzz 40% -fill "#CCCCCC" -opaque "#000000" out.png && convert out.png -fuzz 10% -fill "#222222" -opaque "#FFFFFF" out-2.png`

Output:
![out-2](https://user-images.githubusercontent.com/48932412/100493243-380d7100-3103-11eb-9c28-1210d071a253.png)

As you can see, it doesn't look that nice. I found a command that can take antialiasing into account from [this post](https://www.imagemagick.org/discourse-server/viewtopic.php?p=111241#p111241) on the old forum. 

This is the result I see in my image viewer:
<img width="695" alt="image" src="https://user-images.githubusercontent.com/48932412/100493322-e2859400-3103-11eb-8825-ab640796c37b.png">

This respects antialiasing, but _only_ for text that was originally black. The red and yellow stay the same. Is there any solution for this? 

Note: interestingly enough, if I copy the image from my image viewer and paste it into GitHub directly, I get the following:
![image](https://user-images.githubusercontent.com/48932412/100493304-b36f2280-3103-11eb-8915-9628905ef358.png)

Also, I got these images from the original file I cropped the first image from. When I tried using the first image itself, it gave me an output with a purple background. Not sure what's going on but I presume it has something to do with color spaces (but I'm no expert on color science).

**Edit:**

Maybe what I'm looking for is being able to use the fuzzy color replacement command, but in a way such that the replacement color has the same fuzziness as the color it replaces. So for example, if I'm trying to replace white with black with 10% fuziness, `#FFFFFF` should be changed to `#000000`, but `#EEEEEE` should be replaced to `#111111`. Currently both are changed to `#000000` using the `-fuzz` flag. Is there a feature like this on imagemagick?

**Comment by fmw42** [comment, marked as answer]
try #222222 = rgb(34,34,34) and 34 = 13.3%

`convert img.png -alpha off -colorspace HSI -channel B +level 100x13.3% +channel -colorspace sRGB result.png`


![result](https://user-images.githubusercontent.com/13458218/100496123-7e180400-3106-11eb-9138-e202b42470f0.png)

**Reply by oakif** [reply]
Thank you—this works _incredibly_ well.

**Comment by snibgo** [comment]
@oakif: I'm confused. Do you want to change just the background (from white to #222222), or do you also want to change the nearly-black lines to nearly white?

**Reply by oakif** [reply]
I wanted to both together! I apologize if I was unclear. The current answer does exactly what I was trying to achieve!
